### ISSUE TITLE
Mark days as day off

### ISSUE THREAD

**Issue by simplay** [rathrio/punch#5]
Extend punch such that we easily can mark a day as a _free day_. _Free days_ or _days off_
are not taken into account when computing the statistics.

I'd like to use / have the following two commands:

```bash
punch -d 24.12.2017 --day-off # marks a given date as free day
punch --day-off # marks the current day as free day
```

To achieve this, we could extend the day class as the follows:

```ruby
class Day
  def initialize(date)
    # ...
    @is_day_off = false
  end

  def day_off!
    @day_off = true
  end
  
  def day_off?
    @day_off
  end
end
```
When computing the statistics, we have to filter every day that is marked as being a day_off.
this can be achieved by defining a custom day selector in the month class:

```ruby
class Month
  # maybe, we'd better introduce an additional getter, which returns the relevant working day
  def days
    @days.reject(&:day_off?)
  end
end
```

@rathrio: what's your opinion about this feature?

**Comment by rathrio** [comment]
As discussed, it makes the most sense to adjust the stats that they ignore days with a specific tag (such as "ignore"). This way we do not have to introduce additional command line args.

**Comment by rathrio** [comment]
Resolved by 8d91abe645e8d06617957056bc4818b16a57eafc, fc730d324081c3274a79d3cb977b345896f9bcaa.
